gms/feature_service.cc

```diff
@@ -23,32 +23,24 @@ namespace gms {
 
 static logging::logger logger("features");
 
-feature_config::feature_config() {
-}
-
-feature_service::feature_service(feature_config cfg) : _config(cfg)
-{}
-
-feature_config feature_config_from_db_config(db::config& cfg, std::set<sstring> disabled) {
-    feature_config fcfg;
-
-    fcfg._disabled_features = std::move(disabled);
+static std::set<sstring> disabled_features_from_db_config(db::config& cfg) {
```

Comment:
That's step backwards. The overall direction is that db::config is not seen by any service, instead -- all services have their own configs. Since you want to watch the changes of db::config it should be done like in b112a98318cd572cd8280b265b639dda8b6cad2d -- with the help of `updateable_value`s and `observable`s (probably we'll need to wait for #7316 to get fixed, but extremely like it will)

Reply:
I guess we could create a `feature_service::config` which stores the subset of config variables that `feature_service` depends on in the form of `updateable_value`s.